src/utils/ProfileProvider.tsx

```diff
@@ -117,10 +104,12 @@ export default function ProfileProvider({ children }: { children: ReactNode }) {
      * setup for the profiles table.
      */
     const updateProfile = async (updatedInfo: Partial<Profile>) => {
```

Comment:
can you do something similar with the `useCallback`s here? this is completely my fault - sorry about this.